packages/mobile/src/harmony-native/components/TextLink/TextLink.tsx

```diff
@@ -45,43 +49,71 @@ export const TextLink = <ParamList extends ReactNavigation.RootParamList>(
   const variantPressingColors = {
     default: color.primary.p300,
     subdued: color.primary.p300,
-    visible: color.link.visible,
-    inverted: color.static.white,
+    visible: color.static.white,
+    inverted: color.link.visible,
     active: color.primary.primary
   }
 
   const animatedStyles = useAnimatedStyle(() => ({
     color: interpolateColor(
-      pressed.value,
+      animatedPressed.value,
       [0, 1],
       [variantColors[variant], variantPressingColors[variant]]
     )
   }))
 
-  const element = (
-    <Flex row gap='xs' alignItems='center'>
+  // Need to nest the AnimatedText inside a Text so the handlers & animation work
+  // while still supporting proper Text layout
+  return (
+    <Text suppressHighlighting variant={textVariant} {...other}>
       <AnimatedText
         style={[
           style,
           animatedStyles,
-          { textDecorationLine: showUnderline ? 'underline' : 'none' }
+          {
+            textDecorationLine: showUnderline ? 'underline' : 'none'
+          }
         ]}
-        variant={textVariant}
-        {...other}
       >
         {children}
       </AnimatedText>
+    </Text>
+  )
+}
+
+/**
```

Comment:
agreed that if we can get to a place where there is a single TextLink, that would be great!